zstacklib/zstacklib/utils/qmp.py

```
@@ -43,27 +43,79 @@ def do_yank(vm):
+def _execute_qmp_command(domain_id, command, raise_exception=True):
+    r, o, e = bash.bash_roe("virsh qemu-monitor-command %s '%s'" % (domain_id, qmp_subcmd(QEMU_VERSION, command)))
+    if r == 0:
+        ret = json.loads(o.strip())
+        if "error" not in ret:
+            return ret["return"]
+        err_msg = "Failed to execute qmp command '{}', vmUuid:{}, error:{}".format(command, domain_id, ret["error"])
+    else:
+        err_msg = "Failed to execute qmp command '{}', vmUuid:{}, retcode:{}, stderr:{}".format(command, domain_id, r, e)
+
+    if raise_exception:
+        raise Exception(err_msg)
```

Ruff (0.14.5)

Create your own exception

(TRY002)